File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-same-direction/remove-element/code/java/RemoveElementSol2.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int n = nums.length, i = 0;
        while (i < n) {
            if (nums[i] == val) nums[i] = nums[--n];
            else i++;
        }
        return n;
    }
}

File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-same-direction/remove-element/code/cpp/remove_element_sol2.cpp
class Solution {
public:
    // Fewer writes — swap val with end element
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size(), i = 0;
        while (i < n) {
            if (nums[i] == val) { nums[i] = nums[n-1]; n--; } // Swap with end
            else i++;  // Only advance if current is not val
        }
        return n;
    }
};